common/txmgr/types/storage_service.go

```diff
@@ -0,0 +1,95 @@
+	// // InsertEthReceipt only used in tests. Use SaveFetchedReceipts instead
+	InsertEthReceipt(receipt *Receipt[R, HASH], opts ...any) error
+	// InsertEthTx(etx *TX, opts ...any) error
+	// InsertEthTxAttempt(attempt *TXATTEMPT, opts ...any) error
+	// LoadEthTxAttempts(etx *TX, opts ...any) error
+	// LoadEthTxesAttempts(etxs []*TX, opts ...any) error
+	// MarkAllConfirmedMissingReceipt(chainID CHAINID, opts ...any) (err error)
+	// MarkOldTxesMissingReceiptAsErrored(blockNum int64, finalityDepth uint32, chainID CHAINID, opts ...any) error
+	// PreloadEthTxes(attempts []TXATTEMPT, opts ...any) error
+	PruneUnstartedEthTxQueue(queueSize uint32, subject uuid.UUID, opts ...any) (n int64, err error)
+	// SaveConfirmedMissingReceiptAttempt(ctx context.Context, timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	SaveFetchedReceipts(receipts []R, chainID CHAINID, opts ...any) (err error)
+	// SaveInProgressAttempt(attempt *TXATTEMPT, opts ...any) error
+	// SaveInsufficientEthAttempt(timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	// SaveReplacementInProgressAttempt(oldAttempt TXATTEMPT, replacementAttempt *TXATTEMPT, opts ...any) error
+	// SaveSentAttempt(timeout time.Duration, attempt *TXATTEMPT, broadcastAt time.Time, opts ...any) error
+	// SetBroadcastBeforeBlockNum(blockNum int64, chainID CHAINID, opts ...any) error
+	// UpdateBroadcastAts(now time.Time, etxIDs []TXID, opts ...any) error
+	// UpdateEthKeyNextNonce(newNextNonce, currentNextNonce TXMETA, address ADDR, chainID CHAINID, opts ...any) error
+	// UpdateEthTxAttemptInProgressToBroadcast(etx *TX, attempt TXATTEMPT, NewAttemptState TxAttemptState, incrNextNonceCallback CallbackFunc, opts ...any) error
+	// UpdateEthTxsUnconfirmed(ids []int64, opts ...any) error
+	// UpdateEthTxUnstartedToInProgress(etx *TX, attempt *TXATTEMPT, opts ...any) error
+	// UpdateEthTxFatalError(etx *TX, opts ...any) error
+	// UpdateEthTxForRebroadcast(etx TX, etxAttempt TXATTEMPT, opts ...any) error
+	// Close()
+}
+
+// TxStrategy controls how txes are queued and sent
+//
+//go:generate mockery --quiet --name TxStrategy --output ./mocks/ --case=underscore --structname TxStrategy --filename tx_strategy.go
+type TxStrategy interface {
+	// Subject will be saved to eth_txes.subject if not null
+	Subject() uuid.NullUUID
+	// PruneQueue is called after eth_tx insertion
+	PruneQueue(pruneService any, opt any) (n int64, err error)
+}
+
+// R is the raw transaction receipt
+type ReceiptPlus[R any] struct {
+	ID           uuid.UUID `db:"id"`
+	Receipt      R         `db:"receipt"`
+	FailOnRevert bool      `db:"FailOnRevert"`
+}
+
+type Receipt[R any, HASH any] struct {
+	ID               int64
+	TxHash           HASH
+	BlockHash        HASH
+	BlockNumber      int64
+	TransactionIndex uint
+	Receipt          R
+	CreatedAt        time.Time
+}
+
+type TxAttemptState string
+
+type TxState string
+
+type CallbackFunc func(opts ...any) error
```

Comment:
more comments on what this is, details of the param type.
Perhaps rename the func name to be more descriptive

Reply:
It was meant to be a function that accepts a pg.Queryer. Not needed anymore (for now) because we will have sqlx-related parameters in the generic ORM